src/main/java/com/example/kursovik2/service/ExaminerServiceImpl.java

```diff
@@ -0,0 +1,66 @@
+package com.example.kursovik2.service;
+
+import com.example.kursovik2.exception.BadRequestException;
+import com.example.kursovik2.model.Question;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+@Service
+
+public class ExaminerServiceImpl implements ExaminerService{
+    private final QuestionService javaQuestionService;
+    private final QuestionService mathQuestionService;
+
+
+    public ExaminerServiceImpl(@Qualifier("javaQuestionService") QuestionService javaQuestionService,
+                               @Qualifier("mathQuestionService") QuestionService mathQuestionService) {
+
+        this.javaQuestionService = javaQuestionService;
+        this.mathQuestionService = mathQuestionService;
+
+    }
+    @Override
+    public Collection<Question> getQuestions(int amount) {
+        Collection<Question> questions = new ArrayList<>();
+
+        Collection<Question> javaQuestions = javaQuestionService.getAllQuestions();
+        Collection<Question> mathQuestions = mathQuestionService.getAllQuestions();
```

Comment:
Проблема в этих строках. Нужно вспомнить, чем отличаются ссылочные типы данных от примитивных.
Когда ты используешь тут "=", то ты не создаёшь копию коллекции.
Ты получаешь ссылку на уже существующую.
Поэтому, когда ты пытаешься удалить из неё элементы, ты получаешь ошибку.
Тут тебе нужно явно создать копию коллекции.
Вот самый просто способ:
```suggestion
        Collection<Question> javaQuestions = new ArrayList<>(javaQuestionService.getAllQuestions());
        Collection<Question> mathQuestions = new ArrayList<>(mathQuestionService.getAllQuestions());
```